freight/api/task_index.py

```diff
@@ -137,10 +118,10 @@ def post(self):
             if not sha:
                 return self.error('Invalid ref', name='invalid_ref', status_code=400)
         else:
-            try:
-                sha = vcs_backend.get_sha(ref)
-            except vcs.UnknownRevision:
```

Comment:
was this simply broken? I know we changed this in c64ae92fb8f827626eece01bd21843beba12216e but this always used to work, at the very least. I'd be considering reverting that other change if this was causing problems.

Basically pre-c64ae92fb8f827626eece01bd21843beba12216e we were able to both: deploy master, and deploy [random sha], and it worked just fine.